src/dotnet/APIView/APIViewWeb/Repositories/CosmosReviewRepository.cs

```diff
@@ -33,6 +33,10 @@ public async Task<IEnumerable<ReviewModel>> GetReviewsAsync(bool closed)
                 allReviews.AddRange(result.Resource);
             }
 
+            if(!string.IsNullOrEmpty(Language) && Language != "All")
```

Comment:
we should filter this as part of the query same as we do with IsClosed.

Reply:
Language is not saved in Review model currently. All of existing records has language in revisionmodel which is child of review. But you are right. We need packagename in ReviewModel itself to query and find current master version is approved or not.

Reply:
I think we can use the query for inner objects as well. Cosmos query language should allow it.

Reply:
I have fixed this to refer inner variables using Exists clause